client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx

```diff
@@ -33,3 +82,173 @@ export const professionalEmailCard: ProviderCard = {
 	badge,
 	features: getTitanFeatures(),
 };
+
+const ProfessionalEmailCard: FunctionComponent< EmailProvidersStackedCardProps > = ( props ) => {
+	const { currencyCode = '', hasCartDomain, domain, selectedDomainName, titanMailProduct } = props;
+	const professionalEmail: ProviderCard = professionalEmailCardInformation;
+
+	const isEligibleForFreeTrial = hasCartDomain || isDomainEligibleForTitanFreeTrial( domain );
+
+	const isUpgrading = () => {
+		const { domain, hasCartDomain } = props;
+
+		return ! hasCartDomain && hasEmailForwards( domain );
+	};
+
+	const formattedPriceClassName = classNames( {
+		'professional-email-card__keep-main-price': ! isEligibleForFreeTrial,
+		'professional-email-card__discounted-price': isEligibleForFreeTrial,
+	} );
+
+	const discount = isEligibleForFreeTrial ? <> translate( '3 months free' ) ) </> : null;
+
+	const expandButtonLabel = isUpgrading()
+		? translate( 'Upgrade to %(titanProductName)s', {
+				args: {
+					titanProductName: getTitanProductName(),
+				},
+				comment:
+					'%(titanProductName) is the name of the product, which should be "Professional Email" translated',
+		  } )
+		: translate( 'Add %(titanProductName)s', {
+				args: {
+					titanProductName: getTitanProductName(),
+				},
+				comment:
+					'%(titanProductName) is the name of the product, which should be "Professional Email" translated',
+		  } );
+
+	const [ titanMailbox, setTitanMailbox ] = useState( [
+		buildNewTitanMailbox( selectedDomainName, false ),
+	] );
+	const [ addingToCart, setAddingToCart ] = useState( false );
+	const [ validatedTitanMailboxUuids, setValidatedTitanMailboxUuids ] = useState( [ '' ] );
+	const optionalFields = [ PASSWORD_RESET_TITAN_FIELD, FULL_NAME_TITAN_FIELD ];
+
+	const onTitanConfirmNewMailboxes = () => {
+		const { comparisonContext, domain, hasCartDomain, source } = props;
+
+		const validatedTitanMailboxes = validateTitanMailboxes( titanMailbox, optionalFields );
+
+		const mailboxesAreValid = areAllMailboxesValid( validatedTitanMailboxes, optionalFields );
+		const userCanAddEmail = hasCartDomain || canCurrentUserAddEmail( domain );
+		const userCannotAddEmailReason = userCanAddEmail
+			? null
+			: getCurrentUserCannotAddEmailReason( domain );
+
+		const validatedMailboxUuids = validatedTitanMailboxes.map( ( mailbox ) => mailbox.uuid );
+
+		recordTracksEvent( 'calypso_email_providers_add_click', {
+			context: comparisonContext,
+			mailbox_count: validatedMailboxUuids.length,
+			mailboxes_valid: mailboxesAreValid ? 1 : 0,
+			provider: TITAN_PROVIDER_NAME,
+			source,
+			user_can_add_email: userCanAddEmail,
+			user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
+		} );
```

Comment:
Suggestion: this might make sense as a function prop with a types interface - this means that we can centralize this code in a parent component rather than duplicating the calls.